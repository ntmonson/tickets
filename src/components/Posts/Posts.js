import React from 'react';
import PropTypes from 'prop-types';
import ReactTable from 'react-table';
import reactTableCss from 'react-table/react-table.css';
import withStyles from 'isomorphic-style-loader/lib/withStyles';

const columns = [
  {
    Header: 'Post',
    accessor: 'content',
  },
  {
    Header: 'Created',
    accessor: 'createdAt',
  },
];

// Exported for testing, see https://github.com/kriasoft/react-starter-kit/issues/378
export const Posts = ({ posts }) => (
  <ReactTable
    data={posts}
    columns={columns}
    minRows={0}
    showPageSizeOptions={false}
  />
);

Posts.propTypes = {
  posts: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      content: PropTypes.string.isRequired,
      createdAt: PropTypes.string.isRequired,
      updatedAt: PropTypes.string.isRequired,
    }).isRequired,
  ).isRequired,
};

export default withStyles(reactTableCss)(Posts);
