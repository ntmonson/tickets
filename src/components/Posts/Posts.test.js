/* eslint-env jest */
/* eslint-disable padded-blocks, no-unused-expressions */

import React from 'react';
import renderer from 'react-test-renderer';
import { Posts } from './Posts';

describe('Posts', () => {
  test('Renders table correctly', () => {
    const posts = [
      {
        id: '1234',
        content: 'New post',
        createdAt: 'My new date',
        updatedAt: 'My new updated at',
      },
      {
        id: '12345',
        content: 'New post2',
        createdAt: 'My new date2',
        updatedAt: 'My new updated at2',
      },
    ];
    const component = renderer.create(<Posts posts={posts} />).toJSON();

    expect(component).toMatchSnapshot();
  });
});
