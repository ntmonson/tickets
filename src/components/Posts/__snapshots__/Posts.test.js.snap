// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts Renders table correctly 1`] = `
<div
  className="ReactTable"
  style={Object {}}
>
  <div
    className="rt-table"
    style={undefined}
  >
    <div
      className="rt-thead -header"
      style={
        Object {
          "minWidth": "200px",
        }
      }
    >
      <div
        className="rt-tr"
        style={undefined}
      >
        <div
          className="rt-th  rt-resizable-header -cursor-pointer"
          onClick={[Function]}
          role="heading"
          style={
            Object {
              "flex": "100 0 auto",
              "maxWidth": null,
              "width": "100px",
            }
          }
        >
          <div
            className="rt-resizable-header-content"
          >
            Post
          </div>
          <div
            className="rt-resizer"
            onMouseDown={[Function]}
            onTouchStart={[Function]}
          />
        </div>
        <div
          className="rt-th  rt-resizable-header -cursor-pointer"
          onClick={[Function]}
          role="heading"
          style={
            Object {
              "flex": "100 0 auto",
              "maxWidth": null,
              "width": "100px",
            }
          }
        >
          <div
            className="rt-resizable-header-content"
          >
            Created
          </div>
          <div
            className="rt-resizer"
            onMouseDown={[Function]}
            onTouchStart={[Function]}
          />
        </div>
      </div>
    </div>
    <div
      className="rt-tbody"
      style={
        Object {
          "minWidth": "200px",
        }
      }
    >
      <div
        className="rt-tr-group"
      >
        <div
          className="rt-tr -odd"
          style={undefined}
        >
          <div
            className="rt-td"
            onClick={[Function]}
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            New post
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            My new date
          </div>
        </div>
      </div>
      <div
        className="rt-tr-group"
      >
        <div
          className="rt-tr -even"
          style={undefined}
        >
          <div
            className="rt-td"
            onClick={[Function]}
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            New post2
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            My new date2
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="pagination-bottom"
  >
    <div
      className="-pagination"
      style={undefined}
    >
      <div
        className="-previous"
      >
        <button
          className="-btn"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          Previous
        </button>
      </div>
      <div
        className="-center"
      >
        <span
          className="-pageInfo"
        >
          Page
           
          <div
            className="-pageJump"
          >
            <input
              onBlur={[Function]}
              onChange={[Function]}
              onKeyPress={[Function]}
              type="number"
              value={1}
            />
          </div>
           
          of
           
          <span
            className="-totalPages"
          >
            1
          </span>
        </span>
      </div>
      <div
        className="-next"
      >
        <button
          className="-btn"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          Next
        </button>
      </div>
    </div>
  </div>
  <div
    className="-loading"
  >
    <div
      className="-loading-inner"
    >
      Loading...
    </div>
  </div>
</div>
`;
